### PR TITLE
Setup releases using GitVersion

### LINKED ISSUE #126: move package to semantic versioning
The defacto standard for NuGet package versioning is Semantic Versioning, i.e. 1.2.3 (Major.Minor.Patch) your package https://www.nuget.org/packages/AzurePipelinesToGitHubActionsConverter.Core/ doesn't use this style.

image

It would be better to join other listings and use SemVer.

### LINKED ISSUE #175: Add GitHub Release task
Azure DevOps: https://docs.microsoft.com/en-us/azure/devops/pipelines/tasks/utility/github-release?view=azure-devops

GitHub: https://github.com/actions/create-release

### PR DESCRIPTION
**Pull request by samsmithnz**
- fixes #175 
- fixes #126

### CODE REVIEW

**Comment by samsmithnz** [comment]
@f2calv Look about right?

File: .github/workflows/CI.yml [hunk 1]
```diff
@@ -12,20 +12,29 @@ jobs:
...
     steps:
-    - uses: actions/checkout@v1
-    - name: Custom script to create a unique application version number for this build
-      run: src/BuildVersion.ps1 -ProjectFile "src/AzurePipelinesToGitHubActionsConverter.Core/AzurePipelinesToGitHubActionsConverter.Core.csproj"
-      shell: pwsh
-    - name: Build with dotnet
-      run: dotnet build src/AzurePipelinesToGitHubActionsConverter.Core/AzurePipelinesToGitHubActionsConverter.Core.csproj --configuration Release
+    - uses: actions/checkout@v2
+      with:
+        fetch-depth: 0 #fetch-depth is needed for GitVersion
```
Comment (f2calv) [inline .github/workflows/CI.yml:18]: yes complete fetchDepth is required for gitversion, but that happens by default, so you can delete lines 17 & 18

File: .github/workflows/CI.yml [hunk 2]
```diff
@@ -12,20 +12,29 @@ jobs:
...
+        
+    #Install and calculate the new version with GitVersion  
+    - name: Install GitVersion
+      uses: gittools/actions/gitversion/setup@v0.9.7
+      with:
+        versionSpec: '5.x'
+    - name: Determine Version
```
Comment (f2calv) [inline .github/workflows/CI.yml:25]: just a habit from azdo pipielines, would it not be 'best practise' to have the '- uses: ...' as always the first line of the action (as it's less likely to be customisable) and the 'name: ...' to be the 2nd line?  Also better practise to split all actions with 1 empty line for readability?

File: .github/workflows/CI.yml [hunk 3]
```diff
@@ -12,20 +12,29 @@ jobs:
...
+      uses: gittools/actions/gitversion/execute@v0.9.7
+      id: gitversion # step id used as reference for output values
+    - name: Display GitVersion outputs
+      run: |
+        echo "MajorMinorPatch: ${{ steps.gitversion.outputs.majorMinorPatch }}"
+        echo "NuGetVersionV2: ${{ steps.gitversion.outputs.nuGetVersionV2 }}"
+
```
Comment (f2calv) [inline .github/workflows/CI.yml:32]: as this is ci/cd pipeline, I would add an implicit dotnet restore, and then an implicit dotnet build step (with --no-restore) here, then add --no-build to the dotnet test/pack steps

File: .github/workflows/CI.yml [hunk 4]
```diff
@@ -12,20 +12,29 @@ jobs:
...
+      uses: gittools/actions/gitversion/execute@v0.9.7
+      id: gitversion # step id used as reference for output values
+    - name: Display GitVersion outputs
+      run: |
+        echo "MajorMinorPatch: ${{ steps.gitversion.outputs.majorMinorPatch }}"
+        echo "NuGetVersionV2: ${{ steps.gitversion.outputs.nuGetVersionV2 }}"
+    
+    #Test and pack the code into a NuGet package
     - name: Test with dotnet
       run: dotnet test src/AzurePipelinesToGitHubActionsConverter.Tests/AzurePipelinesToGitHubActionsConverter.Tests.csproj --configuration Release
```
Comment (f2calv) [inline .github/workflows/CI.yml:35]: also, if you mopve your SLN file into the root of the repo, you dont even need to specify the path to the csproj - you can just run 'dotnet test' and it'll run the tests on any test proj in the SLN - so cleaner pipeline

File: .github/workflows/CI.yml [hunk 5]
```diff
@@ -44,3 +53,13 @@ jobs:
     - name: Publish nuget package to nuget.org
       run: dotnet nuget push nugetPackage\*.nupkg --api-key "${{ secrets.GHPackagesToken }}" --source "https://www.nuget.org/"
       shell: pwsh
+    - name: Create Release
```
Comment (f2calv) [inline .github/workflows/CI.yml:56]: add a condition (or the actions equivalent) to only create a new release on PR builds

File: .github/workflows/CI.yml [hunk 6]
```diff
@@ -44,3 +53,13 @@ jobs:
     - name: Publish nuget package to nuget.org
```
Comment (f2calv) [inline .github/workflows/CI.yml:53]: add a condition (or the actions equivalent) to only push a package on PR builds

**Comment by f2calv** [comment]
Another observation which I think is good is to prefix all branch names with their 'category', like;

- refs/heads/feature/ImprovePipeline
- refs/heads/feature/SpeedImprovements
- refs/heads/hotfix/MissionCriticalBug
- refs/heads/bugfix/SomeAnnoyingBug

File: .github/workflows/CI.yml (hunk 5 above)
Comment (samsmithnz) [reply .github/workflows/CI.yml:56]: you mean, only create a new release on the main trunk? I wouldn't want a PR release, only a release off Main. Following?

**Pull request merged**